twill-core/src/main/java/org/apache/twill/internal/TwillRuntimeSpecification.java

```diff
@@ -105,4 +114,17 @@ public String getRmSchedulerAddr() {
   public Map<String, Integer> getMaxRetries() {
     return maxRetries;
   }
+
+  /**
+   * Returns the ZK connection string for the Kafka used for log collections,
+   * or {@code null} if log collection is disabled.
+   */
+  @Nullable
+  public String getKafkaZKConnect() {
+    if (!isLogCollectionEnabled()) {
```

Comment:
How to avoid returning null? You mean by throwing exception so that the caller has to call isLogCollectionEnabled before calling this method? I am not a friend of method coupling like this.